apps/emqx/src/emqx_trace/emqx_trace_formatter.erl

```diff
@@ -166,13 +180,29 @@ to_iolist(Atom) when is_atom(Atom) -> atom_to_list(Atom);
 to_iolist(Int) when is_integer(Int) -> integer_to_list(Int);
 to_iolist(Float) when is_float(Float) -> float_to_list(Float, [{decimals, 2}]);
 to_iolist(SubMap) when is_map(SubMap) -> ["[", kvs_to_iolist(maps:to_list(SubMap)), "]"];
-to_iolist(Char) -> emqx_logger_textfmt:try_format_unicode(Char).
+to_iolist(List) when is_list(List) ->
+    case io_lib:printable_unicode_list(List) of
+        true ->
+            emqx_logger_textfmt:try_format_unicode_to_binary(List);
+        false ->
+            ["[", lists:map(fun to_iolist/1, List), "]"]
+    end;
+to_iolist(Char) ->
+    emqx_logger_textfmt:try_format_unicode_to_binary(Char).
 
 kvs_to_iolist(KVs) ->
     lists:join(
         ", ",
         lists:map(
-            fun({K, V}) -> [to_iolist(K), ": ", to_iolist(V)] end,
+            fun
+                ({K, V}) when
+                    K == ?FORMAT_META_KEY_PACKET
+                ->
+                    %% Already formatted
+                    [to_iolist(K), ": ", V];
```

Comment:
That'a a bit confusing 😅: why only this key(`?FORMAT_META_KEY_PACKET`)?  And why we calculate `to_iolist(?FORMAT_META_KEY_PACKET)` in runtime 🤔

Reply:
regarding `to_iolist(K)`, true, not needed.

as to why, because otherwise it messes up packet logging.

e.g.:

```
2025-09-30T14:52:13.473686+00:00 [MQTT] bah@127.0.0.1:35122 msg: mqtt_packet_sent, packet: [[677978786567754081044328204432680], [659910770108971031156104432821019711511111067111100101610])]
```